### ISSUE TITLE
List Block: Create a list item when users insert a hyphen

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#56173]
## What problem does this address?
When users want to create list items they can do so by typing `-` followed by ` `. Some users don't know that they have to add a space after the `-`.

## What is your proposed solution?
Why not create a list item as soon as as user enters a `-`? Are there any scenarios in which a user would want a `-` at the start of their content where it wasn't followed by a space?

https://github.com/WordPress/gutenberg/assets/275961/907b894d-672f-4dce-99cb-0739840229e0

**Comment by t-hamano** [comment]
My guess is that it follows the rules that require a space between the hyphen and the text in order to represent a list in Markdown.

The Classic Editor also requires a space to convert to a list.

https://github.com/WordPress/gutenberg/assets/54422211/8a5107e5-313a-445c-8345-de7d9539df67

**Comment by ellatrix** [comment]
> Why not create a list item as soon as as user enters a `-`?

Because that would cause more false positives. On the other hand, I'm not sure when you would want a hyphen (or asterisk) at the start of a paragraph. We could try it, and revert if there's too many complaints.

**Comment by ellatrix** [comment]
Btw, a similar comment was made about the `---` shortcut, by @jarekmorawski

**Comment by annezazu** [comment]
I don't think we should do this. It's not a common behavior in Google docs, Pages, Scrivener, or Microsoft Word. If this were a more common pattern, I could see it being worthwhile but, as is, I agree this would create false positives.
